eppoclient/client_test.go

```diff
@@ -132,13 +120,41 @@ func Test_GetStringAssignmentHandlesLoggingPanic(t *testing.T) {
 			},
 		},
 	}
-	mockConfigRequestor.Mock.On("GetConfiguration", "experiment-key-1").Return(config["experiment-key-1"], nil)
 
-	client := newEppoClient(mockConfigRequestor, poller, mockLogger)
+	client := newEppoClient(&configurationStore{flags: config}, nil, nil, mockLogger)
 
 	assignment, err := client.GetStringAssignment("experiment-key-1", "user-1", Attributes{}, "")
 	expected := "control"
 
 	assert.Nil(t, err)
 	assert.Equal(t, expected, assignment)
 }
+
+func Test_client_handlesBanditLoggerPanic(t *testing.T) {
+	var logger = new(mockLogger)
+	logger.Mock.On("LogAssignment", mock.Anything).Return()
+	logger.Mock.On("LogBanditAction", mock.Anything).Panic("logging panic")
+
+	flags := map[string]flagConfiguration{}
+	bandits := map[string]banditConfiguration{
+		"bandit": {
+			BanditKey:    "bandit",
+			ModelName:    "falcon",
+			ModelVersion: "v123",
+			ModelData: banditModelData{
+				Gamma:                  0,
+				DefaultActionScore:     0,
+				ActionProbabilityFloor: 0,
+				Coefficients:           map[string]banditCoefficients{},
+			},
+		},
+	}
+
+	client := newEppoClient(&configurationStore{flags: flags, bandits: bandits}, nil, nil, logger)
+	actions := map[string]ContextAttributes{
+		"action1": {},
+	}
+	client.GetBanditAction("testFlag", "subject", ContextAttributes{}, actions, "bandit")
```

Comment:
we should probably assert what is returned here, making sure it's _not_ the default value.

Reply:
I have added another test case that asserts that.